### PR TITLE
GH-238: Resolve the world-map GeoJSON URL server-side

### PR DESCRIPTION
**Pull request by magicsunday**
Addresses two of the three parts of #238. The payload reduction is deliberately left open — see below.

## Hard-coded module route

```js
const WORLD_GEOJSON_URL =
    "/index.php?route=%2Fmodule%2F_webtrees-statistics_%2FAsset&asset=js/world-map.geojson";
```

webtrees derives the module route name from the **installation directory**, and the README instructs admins to rename the extracted folder. A deviating name makes the route 404, `loadWorldGeoJson()` throws, and the error placeholder renders — so the world map is missing while every other chart works. A partial, silent failure that points nowhere.

This was the only asset URL in the module not going through `assetUrl()`.

## No cache busting

webtrees serves assets with `max-age=31536000` and compensates through the `filemtime` hash `assetUrl()` appends. The hard-coded string had none, so an updated GeoJSON would not reach returning visitors for up to a year. Not hypothetical — the file has been updated once already.

Both follow from the same change: `Module.php` resolves the URL through `assetUrl()`, `page.phtml` carries it on the page root, and `index.js` reads it. The hard-coded string stays as a fallback for the case where the attribute is absent.

## Why the payload reduction is not here

The issue assumed the widget reads one property (`ISO_A2_EH`) and measured a 73.6% reduction on that basis. I grepped the chart library before touching the data — it also reads `NAME`, `NAME_LONG` and `ISO_A2`:

```
properties?.name
ISO_A2   ISO_A2_EH   NAME   NAME_LONG
```

So the measured saving does not hold and the reduction has to be recalculated against the real property set. Doing it on the old figure would have stripped properties the map needs for its tooltips. #238 stays open with this correction.

`composer ci:test` green: 831 PHP tests, 22 JS tests, both PHPStan configs clean, bundle rebuilt, catalogues refreshed.

## Summary by CodeRabbit

* **New Features**
  * World-map statistics now load their GeoJSON data from the configured application asset URL.
  * GeoJSON assets are served with the correct `application/geo+json` content type.
  * A fallback URL is used when no configured asset URL is available.

* **Bug Fixes**
  * Improved reliability when loading world-map data across different deployments.

* **Tests**
  * Added coverage verifying configured URLs, fallback behavior, and GeoJSON asset responses.

* **Documentation**
  * Updated translation reference metadata and world-map configuration guidance.

### CODE REVIEW

**Comment by magicsunday** [comment]
@codex review

File: src/Module.php [hunk 1]
```diff
@@ -196,6 +200,11 @@ public function getChartAction(ServerRequestInterface $request): ResponseInterfa
                 'hero'       => $statistic->getHeroStats(),
                 'javascript' => $this->assetUrl('js/statistics-' . self::CUSTOM_VERSION . '.min.js'),
                 'stylesheet' => $this->assetUrl('css/statistics.css'),
+                // Resolved server-side: the module's route name derives from
+                // the installation directory, which the admin chooses, and
+                // assetUrl() appends a filemtime hash so an updated GeoJSON
+                // reaches returning visitors despite the one-year cache header.
+                'geoJsonUrl' => $this->assetUrl('js/world-map.geojson'),
                 // Pre-resolved asset URLs for the two web-font families
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Module.php:208]: Rejected — the comma is there.

`src/Module.php:207` reads:

```php
                'geoJsonUrl' => $this->assetUrl('js/world-map.geojson'),
```

Verified rather than assumed: `php -l src/Module.php` → *No syntax errors detected*. A missing comma there would also have failed `composer ci:test:php:lint` (240 files, green), PHPStan at level max, and all 811 unit tests — none of which can run against a file that does not parse.

Most likely the diff hunk was read with the four intervening comment lines collapsed.

File: src/Module.php [hunk 2]
```diff
@@ -196,6 +200,11 @@ public function getChartAction(ServerRequestInterface $request): ResponseInterfa
                 'hero'       => $statistic->getHeroStats(),
                 'javascript' => $this->assetUrl('js/statistics-' . self::CUSTOM_VERSION . '.min.js'),
                 'stylesheet' => $this->assetUrl('css/statistics.css'),
+                // Resolved server-side: the module's route name derives from
+                // the installation directory, which the admin chooses, and
+                // assetUrl() appends a filemtime hash so an updated GeoJSON
+                // reaches returning visitors despite the one-year cache header.
+                'geoJsonUrl' => $this->assetUrl('js/world-map.geojson'),
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Module.php:207]: Agreed, and already tracked: #291, filed from the same observation in my own review pass before your comment landed.

To be precise about what is and is not covered here — the consumer end *is* tested in this PR (`resources/js/tests/dispatcher.test.js`, three cases asserting which URL is fetched for attribute-present / absent / empty, verified to red when `worldGeoJsonUrl()` ignores the attribute). What is missing is the producer end: nothing asserts that `page.phtml` actually emits `data-geojson-url`. So a rename there stays green while production silently falls back to the hard-coded route — the exact failure you are pointing at.

It sits in its own issue rather than this PR because rendering the page shell needs `$tree`, `$tabs`, `$module`, `$title`, `$hero` (a `HeroStats` DTO), `$javascript`, `$stylesheet` and `$fontUrls` supplied first. The house pattern for it exists (`tests/Integration/TreeHealthPartnershipReachViewTest.php` renders a module view through the real `view()` and asserts emitted `data-*` attributes), so #291 carries the concrete assertion to write.
